enhancements/installer/openstack-availibility-zones.md

```diff
@@ -0,0 +1,125 @@
+
+## Open Questions
+
+- Will the node's AZ be used for its volume? Will this effect the behavior of the cluster?
+
+## Summary
+
+The installer should automatically discover all availability zones (AZs) and distribute the control plane and compute nodes across them to maximize availibility by default. To accomodate use cases with AZ restrictions, the installer should also allow users to specify a set of them for each machinepool.
+
+## Motivation
+
+### Goals
+
+- Enable the discovery of AZs in OpenStack
+- Spread out control plane and machine nodes across AZs to maximize availibility as a default behavior
+- Add an option to restrict a machine pool to a set of AZs in the installer
+
+### Non-Goals
+
+### Proposal
+
+### User Stories
+
+#### Day 2 Additional MachineSet
+
+As a user, I want to be able to add Machinepools on a different AZ then the one the installer is currently installed on in order to increase the availibility of my cluster, and to re-distribute the load.
+
+#### Install time Machinepool AZ customization
+
+As a user, I want to customize which AZs each MachineSet can be installed onto when the cluster is first installed.
+
+#### Default AZ discovery and HA installation
+
+When I install OpenShift on an OpenStack cluster that has multiple AZs, I want the installer to discover the available AZs and install the cluster across all of them in a way that is highly available without taking away from the user experience.
+
+## Implementation Details/Notes/Constraints
+
+This comes down to the following core features:
+1. allowing custom AZs for Machinepools
+2. enabling the installer to lookup AZs
+3. creating a default HA deployment across discovered AZs
+
+### Custom AZs for Machinepools
+
+This feature would allow users to specify a list of AZs that the installer should create nodes on for a given Machinepool. It will attempt to evenly distribute load across all listed AZs. The user interface should match the [AWS implementation](https://github.com/openshift/installer/blob/master/docs/user/aws/customization.md#machine-pools), for the best user experience across platforms.
+
+### AZ lookup
+
+The installer should have the ability to discover AZs in a given OpenStack cloud and cache this information. This will be useful for validating the AZs passed to the Machinepools, and for selecting which AZs to install onto by default. This would not require any changes to the user interface, and should take nothing more than an API query.
+
+### Default HA Deployment
+
+When the installer is run without explicit AZs provided, it should reference the AZs it discovered and spread the install out over them. Users that wish to narrow the scope of this can specify custom AZs as noted above.
+
+### Risks and Mitigations
+
+- If persistent volume claims cannot be made against volumes in different AZs, then that would be a huge user experience issue, and could delay or block the feature.
+
```

Comment:
We could add as a risk that before this enhancement, Nova would have only chosen Availability zones with enough resources. With this enhancement, by default the Installer will force the creation of instances on all available AZs, regardless of whether they have capacity to host the servers.